Sources/MotorControlUI/TappingStepView.swift

```diff
@@ -0,0 +1,260 @@
+import MobilePassiveData
+import MotorControl
+import SharedMobileUI
+import SharedResources
+
+struct TappingStepView: View {
+    @EnvironmentObject var assessmentState: AssessmentState
+    @EnvironmentObject var pagedNavigation: PagedNavigationViewModel
+    @SwiftUI.Environment(\.surveyTintColor) var surveyTint: Color
+    @SwiftUI.Environment(\.spacing) var spacing: CGFloat
+    @ObservedObject var state: TappingStepViewModel
+    @State var timer = Timer.publish(every: 1, on: .main, in: .common).autoconnect()
+
+    @ViewBuilder
+    fileprivate func insideCountdownDial(_ count: Int) -> some View {
+        VStack {
+            Text("\(count)")
+                .font(.countdownNumbers)
+                .foregroundColor(.textForeground)
+                .frame(maxWidth: .infinity, alignment: .center)
+            Text("seconds", bundle: SharedResources.bundle)
+                .font(.countdownDialText)
+                .foregroundColor(.textForeground)
+        }
+    }
+    
+    @ViewBuilder
+    fileprivate func countdownDial() -> some View {
+        ZStack {
+            insideCountdownDial(Int(state.countdown))
+            insideCountdownDial(30)
+                .opacity(0)
+        }
+        .fixedSize(horizontal: true, vertical: true)
+        .padding(48)
+        .background (
+            Circle()
+                .trim(from: 0.0, to: min(state.progress, 1.0))
+                .stroke(style: StrokeStyle(lineWidth: 5, lineCap: .round))
+                .foregroundColor(.textForeground)
+                .rotationEffect(Angle(degrees: 270.0))
+                .padding(2.5)
+                .pausableAnimation(progress: $state.progress,
+                                   paused: $state.isPaused,
+                                   remainingDuration: $state.countdown)
+                .background (
+                    Circle()
+                        .fill(Color.sageWhite)
+                )
+        )
+    }
+    
+    @ViewBuilder
+    fileprivate func singleTappingButton(target: TappingButtonIdentifier) -> some View {
+        Text("Tap", bundle: SharedResources.bundle)
+            .frame(width: 100, height: 100)
+            .foregroundColor(Color.textForeground)
+            .background(surveyTint.saturation(2))
+            .clipShape(Circle())
+            .onFingerPressedGesture { startLocation, tapDuration in
+                guard state.recorder.clock.runningDuration() < state.motionConfig.duration else { return }
+                state.tappedScreen(uptime: SystemClock.uptime(),
+                                   timestamp: state.recorder.clock.runningDuration(),
+                                   currentButton: target, location: startLocation,
+                                   duration: tapDuration)
+            }
+            .simultaneousGesture(
+                DragGesture(minimumDistance: 0)
+                    .onChanged { touch in
+                        guard !state.initialTap else { return }
```

Comment:
You raised a good point about this flag where b/c starting the recorder is an async task, this could be called more than once. Maybe change it back?